app/src/main/java/com/zeapo/pwdstore/UserPreference.kt

```diff
@@ -519,9 +520,39 @@ class UserPreference : AppCompatActivity() {
         startActivityForResult(intent, SET_CUSTOM_XKPWD_DICT)
     }
 
-    @Throws(IOException::class)
+    @Throws(IllegalArgumentException::class, IOException::class)
     private fun copySshKey(uri: Uri) {
-        // TODO: Check if valid SSH Key before import
+        // See metadata from document to validate SSH key
+        contentResolver.query(
+                uri, null, null, null, null, null
+        )?.use { cursor ->
+            val sizeIndex = cursor.getColumnIndex(OpenableColumns.SIZE)
+
+            // cursor returns only 1 row
+            cursor.moveToFirst()
+            // see file's metadata
+            val sizeFile = cursor.getInt(sizeIndex)
+
+            // SSH key size < than 100KB and different from 0
+            if (sizeFile > 100000 ||
+                    sizeFile == 0)
+                throw IllegalArgumentException("Wrong file type selected")
+            else {
+                // Validate BEGIN and END markers
+                val lines = contentResolver.openInputStream(uri)?.bufferedReader()?.readLines()
+
+                // file must have more than 2 lines
+                // first line contains BEGIN
+                // last line contains END
+                if (lines != null &&
+                    lines.size > 2 &&
+                    !lines[0].contains("BEGIN") &&
```

Comment:
```suggestion
                    !lines[0].contains("BEGIN OPENSSH PRIVATE KEY") &&
```

Reply:
I don't think it should be "BEGIN OPENSSH PRIVATE KEY"
since different types of keys have different texts. 
I tried with rsa, dsa, ed25519 and ecdsa. They have the structure of `BEGIN XXXXX PRIVATE KEY`

We could have 2 validations if you want to confirm the whole text is in the line, something like
`!lines[0].contains("BEGIN") && !lines[0].contains("PRIVATE KEY")`

Reply:
Yeah that was my bad, I did not confirm with RSA keys. @FabianHenneke can you fold in this change in your PR as well? Also update the changelog to mention this. Thanks!